From b27291b7ce8107d0c6d63f1dcaf41ad451db4d53 Mon Sep 17 00:00:00 2001
From: z00512904 <z00512904@huawei.com>
Date: Tue, 29 Nov 2022 13:57:31 +0000
Subject: [PATCH] oom: add oom notifier call for oom panic
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: b27291b7ce8107d0c6d63f1dcaf41ad451db4d53
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I63SDZ

-------------------------------

When cause oom and enable CONFIG_ASCEND_OOM and oom_panic, we should call
oom_type_notifier_call() to tell others this message.

Signed-off-by: z00512904 <z00512904@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 mm/oom_kill.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/oom_kill.c b/mm/oom_kill.c
index 417ff9574d19..0f77eb4c6644 100644
--- a/mm/oom_kill.c
+++ b/mm/oom_kill.c
@@ -1091,6 +1091,7 @@ static void check_panic_on_oom(struct oom_control *oc)
 	if (is_sysrq_oom(oc))
 		return;
 	dump_header(oc, NULL);
+	oom_type_notifier_call(0, oc);
 	panic("Out of memory: %s panic_on_oom is enabled\n",
 		sysctl_panic_on_oom == 2 ? "compulsory" : "system-wide");
 }
-- 
2.33.0

